cmd/influx/write_test.go

```diff
@@ -120,24 +156,191 @@ func Test_writeFlags_createLineReader(t *testing.T) {
 		// output
 		firstLineCorrection int // 0 unless shifted by prepended headers or skipped rows
 		lines               []string
-		// lpData indicates the the data are line protocol data
-		lpData bool
 	}{
+		{
+			name: "read data from LP file",
+			flags: writeFlagsBuilder{
+				Files: []string{lpFile},
+			},
+			firstLineCorrection: 0,
+			lines: []string{
+				lpContents,
+			},
+		},
+		{
+			name: "read data from LP file using non-UTF encoding",
+			flags: writeFlagsBuilder{
+				Files:    []string{lpFile},
+				Encoding: "ISO_8859-1",
+			},
+			firstLineCorrection: 0,
+			lines: []string{
+				lpContents,
+			},
+		},
+		{
+			name: "read compressed LP data from file",
+			flags: writeFlagsBuilder{
+				Files:       []string{gzipLpFileNoExt},
+				Compression: inputCompressionGzip,
+			},
+			firstLineCorrection: 0,
+			lines: []string{
+				lpContents,
+			},
+		},
+		{
+			name: "read compressed data from LP file using non-UTF encoding",
+			flags: writeFlagsBuilder{
+				Files:       []string{gzipLpFileNoExt},
+				Compression: inputCompressionGzip,
+				Encoding:    "ISO_8859-1",
+			},
+			firstLineCorrection: 0,
+			lines: []string{
+				lpContents,
+			},
+		},
+		{
+			name: "read compressed LP data from file ending in .gz",
+			flags: writeFlagsBuilder{
+				Files: []string{gzipLpFile},
+			},
+			firstLineCorrection: 0,
+			lines: []string{
+				lpContents,
+			},
+		},
+		{
+			name: "read compressed and uncompressed LP data from file in the same call",
+			flags: writeFlagsBuilder{
+				Files: []string{gzipLpFile, lpFile},
+			},
+			firstLineCorrection: 0,
+			lines: []string{
+				lpContents,
+				lpContents,
+			},
+		},
+		{
+			name:  "read LP data from stdin",
+			flags: writeFlagsBuilder{},
+			stdIn: strings.NewReader(stdInLpContents),
+			lines: []string{
+				stdInLpContents,
+			},
+		},
+		{
+			name: "read compressed LP data from stdin",
+			flags: writeFlagsBuilder{
+				Compression: inputCompressionGzip,
+			},
+			stdIn: stdInLpGzipContents,
+			lines: []string{
+				stdInLpContents,
+			},
+		},
+		{
+			name:      "read LP data from stdin using '-' argument",
+			flags:     writeFlagsBuilder{},
+			stdIn:     strings.NewReader(stdInLpContents),
+			arguments: []string{"-"},
+			lines: []string{
+				stdInLpContents,
+			},
+		},
+		{
+			name: "read compressed LP data from stdin using '-' argument",
```

Comment:
Ugh, this one passes if run on its own, but fails when all tests run... investigating

Reply:
Fixed (locally at least)